rclcpp/test/rclcpp/test_duration.cpp

```diff
@@ -106,6 +107,30 @@ TEST_F(TestDuration, operators_with_message_stamp) {
 
   EXPECT_THROW(neg_time_msg + max, std::runtime_error);
   EXPECT_THROW(time_msg + max, std::overflow_error);
+
+  // Addition and subtraction assignment operators
+  time_msg = rclcpp::Time(0, 100000000u);
+  time_msg += pos_duration;
+  EXPECT_EQ(time_msg.sec, 1);
+  EXPECT_EQ(time_msg.nanosec, 200000000u);
+
+  time_msg -= pos_duration;
```

Comment:
Any special handling needed for overflow/underflow?

Reply:
Hi @Ryanf55,
Sorry for forgetting to reply.
The overflow test is already in line 133 below.
For the underflow test, please check here: https://github.com/ros2/rclcpp/pull/2419#issuecomment-1918304478.

Or let me paraphrase in another way:
I follow what happens when `rclcpp::Duration + msg::Time`.
It first becomes `rclcpp::Duration + rclcpp::Time` (implicit conversion).
During the conversion, negative `seconds` in `msg::Time` is not allowed.
https://github.com/ros2/rclcpp/blob/99f0fc938b2d0c441787113a3920c16d3614a197/rclcpp/src/rclcpp/time.cpp#L52-L54
IMO this is a small design flaw since it does not match Unix that uses `int32` to store time.
But in practice, who cares about time before 1970, so it should remain fine.
After that, positive `int32` seconds and `uint32` nanoseconds will generate positive `int64` full nanoseconds.
A positive `int64` nanoseconds add another `int64` duration nanoseconds always not underflow.

That's my current understanding and therefore I do (and can) not test underflow.

Reply:
Is it written down anywhere that `rclcpp::Time` must use Unix time? I have used it in places to store time since boot, in the case that the embedded device does not have a RTC or network connection, and thus had no idea of any time except boot time. Certain unsafe uses of while loops, with timeouts and such can lead to weird bugs. 

https://docs.ros2.org/beta3/api/rclcpp/classrclcpp_1_1Time.html

Reply:
> Is it written down anywhere that `rclcpp::Time` must use Unix time?
I don't find such description, maybe my statement earlier is too biased.
> I have used it in places to store time since boot, in the case that the embedded device does not have a RTC or network connection, and thus had no idea of any time except boot time.
Yes, this reminds me that the clock source of `rclcpp::Time` could also be timestamps in rosbag.
Both of them are not using current epoch time.

Reply:
As long as overflow/overflow is tested somewhere, seems like a good approach to me.